packages/yew/src/functional/hooks/use_ref.rs

```diff
@@ -47,25 +50,9 @@ use std::{cell::RefCell, rc::Rc};
 ///     }
 /// }
 /// ```
+#[hook]
 pub fn use_mut_ref<T: 'static>(initial_value: impl FnOnce() -> T) -> Rc<RefCell<T>> {
-    use_hook(
-        || Rc::new(RefCell::new(initial_value())),
-        |state, _| state.clone(),
-        |_| {},
-    )
-}
-
-/// This hook is used for obtaining a immutable reference to a stateful value.
-/// Its state persists across renders.
-///
-/// If you need a mutable reference, consider using [`use_mut_ref`](super::use_mut_ref).
-/// If you need the component to be re-rendered on state change, consider using [`use_state`](super::use_state()).
-pub fn use_ref<T: 'static>(initial_value: impl FnOnce() -> T) -> Rc<T> {
-    use_hook(
-        || Rc::new(initial_value()),
-        |state, _| Rc::clone(state),
-        |_| {},
-    )
+    use_memo(|_| RefCell::new(initial_value()), ())
```

Comment:
I do not agree with this.
`use_ref` is a way to store data for a components instance that when changed does not cause a re-render.
`use_memo` is a `use_state` just that updates automatically based on dependencies.

React has same reasoning and thus it has both hooks in the base library.

please undo the removal of old `use_ref`

Reply:
Ok actually i see might be overreacting.
you only removed the non-mut version of use ref. Ans if we update `use_mut_ref` it will not cause a rerender, so my bad.